zebra-chain/src/orchard/tests/preallocate.rs

```diff
@@ -0,0 +1,95 @@
+//! Tests for trusted preallocation during deserialization.
+
+use crate::{
+    block::MAX_BLOCK_BYTES,
+    orchard::{
+        shielded_data::{ACTION_SIZE, SPEND_AUTH_SIG_SIZE},
+        Action,
+    },
+    primitives::redpallas::{Signature, SpendAuth},
+    serialization::{TrustedPreallocate, ZcashSerialize},
+};
+
+use proptest::{prelude::*, proptest};
+use std::convert::TryInto;
+
+proptest! {
+    /// Confirm that each Action takes exactly ACTION_SIZE bytes when serialized.
+    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
+    #[test]
+    fn action_size_is_correct(action in <Action>::arbitrary_with(())) {
+        let serialized = action.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
+        prop_assert!(serialized.len() as u64 == ACTION_SIZE)
+    }
+    /// Confirm that each Signature<SpendAuth> takes exactly SPEND_AUTH_SIG_SIZE bytes when serialized.
+    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
+    #[test]
+    fn spend_auth_sig_size_is_correct(sig in <Signature<SpendAuth>>::arbitrary_with(())) {
+        let serialized = sig.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
+        prop_assert!(serialized.len() as u64 == SPEND_AUTH_SIG_SIZE)
+    }
```

Comment:
There's no need to do tests on `Signature<SpendAuth>`, because it uses the same size as `Action`, so the tests for `Action::max_allocation` cover it.